libvast/test/system/archive.cpp

```diff
@@ -42,9 +43,25 @@ struct fixture : fixtures::deterministic_actor_system_and_events {
     run();
   }
 
+  std::vector<event> query(const ids& ids) {
+    bool done = false;
+    std::vector<event> result;
+    self->send(a, ids);
+    run();
+    self->do_receive(
+      [&](vast::system::done_atom, const caf::error& err) {
+        REQUIRE(!err);
+        done = true;
+      },
+      [&](table_slice_ptr slice) {
+        to_events(result, *slice, ids);
+      }
+    ).until([&]{ run(); return done; });
```

Comment:
We already call `run()` before the loop. Since this will run everything until no activity remains, there's no need to call it repeatedly.